distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Storage.enso

```diff
@@ -64,11 +68,18 @@ closest_storage_type value_type = case value_type of
     Value_Type.Mixed -> AnyObjectType.INSTANCE
     Value_Type.Decimal _ scale ->
         is_integer = scale.is_nothing || scale <= 0
-        if is_integer then BigIntegerType.INSTANCE else
-            Error.throw (Illegal_Argument.Error "Columns of type "+value_type.to_display_text+" are currently not supported in the in-memory backend - only Decimal of integer type (scale <= 0) is supported. You may cast the column to Float first (lossy conversion).")
+        if is_integer then BigIntegerType.INSTANCE else BigDecimalType.INSTANCE
     _ ->
         Error.throw (Illegal_Argument.Error "Columns of type "+value_type.to_display_text+" are currently not supported in the in-memory backend.")
 
+enso_to_java x = case x of
+    Decimal.Value big_decimal -> big_decimal
+    _ -> x
+
+java_to_enso x = case x of
+    _ : BigDecimal -> Decimal.Value x
```

Comment:
`java_to_enso` is a conversion function that wraps `java.math.BigDecimal` into Enso's `Decimal` by calling the `Value` constructor. That's correct wrapping. What kind of problem is associated with it? Are you searching for alternatives?

You can call the factory in Java, if you want. If you create a method:
```java
public static Object factory(Function<BigDecimal,Object> f) {
  return f.apply(BigDecimal.valueOf(1));
}
```
and call it from Enso as
```ruby
IO.println <| factory Decimal.Value
```
then you should get `Decimal.Value 1`. Other than this, I am not sure what to do/change.